presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkMemoryBasedBroadcastDependency.java

```diff
@@ -40,17 +44,23 @@ public class PrestoSparkMemoryBasedBroadcastDependency
     private final long queryCompletionDeadline;
     private Broadcast<List<PrestoSparkSerializedPage>> broadcastVariable;
     private final Set<PrestoSparkServiceWaitTimeMetrics> waitTimeMetrics;
+    private final boolean isNativeExecution;
 
-    public PrestoSparkMemoryBasedBroadcastDependency(RddAndMore<PrestoSparkSerializedPage> broadcastDependency, DataSize maxBroadcastSize, long queryCompletionDeadline, Set<PrestoSparkServiceWaitTimeMetrics> waitTimeMetrics)
+    public PrestoSparkMemoryBasedBroadcastDependency(RddAndMore<PrestoSparkSerializedPage> broadcastDependency,
```

Comment:
I am bit confused here. Shouldn't this code live inside `PrestoSparkStorageBasedBroadcastDependency`? `PrestoSparkMemoryBasedBroadcastDependency` is used when HT is directly collected on Driver side and broadcasted while StorageBasedBroadcastDependency is used when HT is written to storage and only FileHandles information is collected on driver.

Reply:
The naming of this class assumes that when we get presto page, it will be in memory broadcast. While for Sapphire-Velox, we return broadcast info as presto pages.

Reply:
I have  moved out logic to dedicated class PrestoSparkNativeStorageBasedBroadcastDepency